### PR TITLE
Modified usages of sets.String{} to use the new generic type all across the kubernetes scheduler.(Waiting for generics to be part of the current supported releases to merge))

### LINKED ISSUE #113709: Replace all deprecated `sets.String` with `sets.Set`
In this PR https://github.com/kubernetes/kubernetes/pull/112377, `sets.Set` is newly implemented, and `sets.String` got marked as deprecated.

In the scheduler, we often use `sets.String` everywhere. This issue aims to replace all `sets.String` (+ other deprecated structs in the `sets` package) with `sets.Set` in the sig/scheduling related area.

You can refer to the following comments.
https://github.com/kubernetes/kubernetes/blob/7158e0a4ffffcf50ce7e2f6de0c25fbb414b62b0/staging/src/k8s.io/apimachinery/pkg/util/sets/string.go#L23-L24

/kind cleanup
/sig scheduling
/good-first-issue

The sig/scheduling related area (I'm not sure if all of them use `sets.String` in it.):

 - [ ] `pkg/scheduler` 
 - [ ] `test/e2e/scheduling`
 - [ ] `test/integration/scheduler`
 - [ ] `test/integration/scheduler_perf`
 - [ ] `cmd/kube-scheduler`

### PR DESCRIPTION
**Pull request by Bryce-Soghigian**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
WE deprecated sets.String to be replaced with the [generic](https://github.com/kubernetes/kubernetes/pull/112377) implementation of Set.

This PR is for the sig-scheduling cleanup and moving over to the new generic types.
#### Which issue(s) this PR fixes:

Fixes #113709

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: pkg/scheduler/internal/queue/scheduling_queue.go [hunk 1]
```diff
@@ -191,8 +191,8 @@ type priorityQueueOptions struct {
 	podMaxBackoffDuration             time.Duration
 	podMaxInUnschedulablePodsDuration time.Duration
 	podNominator                      framework.PodNominator
-	clusterEventMap                   map[framework.ClusterEvent]sets.String
 	preEnqueuePluginMap               map[string][]framework.PreEnqueuePlugin
+	clusterEventMap                   map[framework.ClusterEvent]sets.Set[string]
```
Comment (Aut0R3V) [inline pkg/scheduler/internal/queue/scheduling_queue.go:195]: I think there are some checks that have not been approved. Could you check them out?

Comment (Bryce-Soghigian) [reply pkg/scheduler/internal/queue/scheduling_queue.go:195]: What do you mean checks? I ran go test here and it passed.

**Review by sanposhiho — COMMENTED** [review]
Please restore removed files under `/vendor`. I guess you removed them mistakenly.

**Comment by kerthcet** [comment]
@Bryce-Soghigian Plz fix the CI first, thanks a lot.

**Comment by Bryce-Soghigian** [comment]
@kerthcet  @sanposhiho @logicalhan I have addressed all of the CI problems. Can i get a LGTM label and or an Approval label?

File: pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go [hunk 2]
```diff
@@ -1734,8 +1734,14 @@ func TestPreempt(t *testing.T) {
 			if len(deletedPodNames) != len(test.expectedPods) {
 				t.Errorf("expected %v pods, got %v.", len(test.expectedPods), len(deletedPodNames))
 			}
-			if diff := cmp.Diff(patchedPodNames.List(), deletedPodNames.List()); diff != "" {
-				t.Errorf("unexpected difference in the set of patched and deleted pods: %s", diff)
+			patchedPodNamesList := getListOfZones(patchedPodNames)
```
Comment (kerthcet) [inline pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go:1737]: Use sets.List() instead.

File: pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go [hunk 3]
```diff
@@ -1734,8 +1734,14 @@ func TestPreempt(t *testing.T) {
...
+			deletedPodNamesList := getListOfZones(deletedPodNames)
+
+			if !patchedPodNames.HasAll(deletedPodNamesList...) {
```
Comment (kerthcet) [inline pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go:1740]: I think it's better to use cmp.Diff()

File: test/e2e/scheduling/ubernetes_lite.go [hunk 4]
```diff
@@ -132,9 +132,9 @@ func SpreadServiceOrFail(f *framework.Framework, replicaCount int, zoneNames set
 	selector := labels.SelectorFromSet(labels.Set(map[string]string{"service": serviceName}))
 	pods, err := e2epod.WaitForPodsWithLabelScheduled(f.ClientSet, f.Namespace.Name, selector)
 	framework.ExpectNoError(err)
-
+	zones := getListOfZones(zoneNames)
 	// Now make sure they're spread across zones
-	checkZoneSpreading(f.ClientSet, pods, zoneNames.List())
+	checkZoneSpreading(f.ClientSet, pods, zones)
```
Comment (kerthcet) [inline test/e2e/scheduling/ubernetes_lite.go:137]: ditto.

File: test/e2e/scheduling/ubernetes_lite.go [hunk 5]
```diff
@@ -236,6 +236,15 @@ func SpreadRCOrFail(f *framework.Framework, replicaCount int32, zoneNames sets.S
...
 	framework.ExpectNoError(err)
 
+	zones := getListOfZones(zoneNames)
 	// Now make sure they're spread across zones
-	checkZoneSpreading(f.ClientSet, pods, zoneNames.List())
+	checkZoneSpreading(f.ClientSet, pods, zones)
+}
+
+// getListOfZones converts a zones set to a list of strings.
+func getListOfZones(zones sets.Set[string]) (output []string) {
```
Comment (kerthcet) [inline test/e2e/scheduling/ubernetes_lite.go:245]: Remove this.

File: test/e2e/storage/ubernetes_lite_volumes.go [hunk 6]
```diff
@@ -147,3 +149,11 @@ func PodsUseStaticPVsOrFail(f *framework.Framework, podCount int, image string)
 		framework.ExpectNoError(err)
 	}
 }
+
+// getListOfZones converts a zones set to a list of strings.
```
Comment (kerthcet) [inline test/e2e/storage/ubernetes_lite_volumes.go:153]: ditto

File: pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go (hunk 2 above)
Comment (Bryce-Soghigian) [reply pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go]: modified code to use sets.List()

**Comment by pohly** [comment]
Change looks good, but needs to be rebased.

**Comment by Bryce-Soghigian** [comment]
@kerthcet  I am looking for that LGTM label again. WE waited for generics to be supported, but it seems that all supported k8s versions right now are using go 1.18 so we can probably merge this in

**Comment by kerthcet** [comment]
fix the tests first?

**Comment by kidddddddddddddddddddddd** [comment]
@Bryce-Soghigian are you coming back for this?

**Comment by kerthcet** [comment]
Kindly ping @Bryce-Soghigian , would you mind to revisit this again? Or if you do not have bandwidth, the other contributors from the community can help with this and make you a co-author, for we have several PRs want to address the same issue.

**Comment by Bryce-Soghigian** [comment]
> Kindly ping @Bryce-Soghigian , would you mind to revisit this again? Or if you do not have bandwidth, the other contributors from the community can help with this and make you a co-author, for we have several PRs want to address the same issue.

Its ok other people can put in their prs

**Comment by kerthcet** [comment]
kindly ping @yoongon @kidddddddddddddddddddddd @abhishekkhanna2710 who volunteers to address the issue, also plz make @Bryce-Soghigian a co-author as his great effort.

**Pull request closed without merging**